### PR TITLE
#15: False "Invalid URL" message

### PR DESCRIPTION
**Pull request by pgpathan22**
#15

### CODE REVIEW

File: app/src/main/java/space/taran/arkshelf/presentation/searchedit/SearchEditFragment.kt [hunk 1]
```diff
@@ -121,6 +122,12 @@ class SearchEditFragment : Fragment(R.layout.fragment_search_edit) {
         is NoInternetException -> {
             binding.inputUrl.error = getString(R.string.no_internet)
         }
+        is ProtocolException -> {
+            binding.inputUrl.error =
+                if (exception.message?.contains("Too many follow-up requests",ignoreCase = true) == true)
```
Comment (kirillt) [inline app/src/main/java/space/taran/arkshelf/presentation/searchedit/SearchEditFragment.kt]: Maybe possible to extract HTTP response code?
Can it be 408?
https://stackoverflow.com/questions/48595358/getting-too-many-follow-up-request-as-response-of-request

Comment (pgpathan22) [reply app/src/main/java/space/taran/arkshelf/presentation/searchedit/SearchEditFragment.kt]: I have tested with the URL but now it is working fine. So can't test it. I will check after some time.

Comment (pgpathan22) [reply app/src/main/java/space/taran/arkshelf/presentation/searchedit/SearchEditFragment.kt]: image

I have checked and it shows 307 (Temporary Redirect) not 408. So the message is right for this event and the stackoverflow post is almost 5 years old. Therefore, we can't trust it either.

Comment (kirillt) [reply app/src/main/java/space/taran/arkshelf/presentation/searchedit/SearchEditFragment.kt]: Ok, but would be good to check HTTP response and log it.
If it is not 2XX (successful response), then might be also worth of including it into message to the user.

Comment (pgpathan22) [reply app/src/main/java/space/taran/arkshelf/presentation/searchedit/SearchEditFragment.kt]: Done

**Review by kirillt — APPROVED** [review]

**Pull request merged**